database/build/clean_test.go

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2023 Target Brands, Inc. All rights reserved.
+//
+// Use of this source code is governed by the LICENSE file in this repository.
+
+package build
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/DATA-DOG/go-sqlmock"
+)
+
+func TestBuild_Engine_CleanBuilds(t *testing.T) {
+	// setup types
+	_buildOne := testBuild()
+	_buildOne.SetID(1)
+	_buildOne.SetRepoID(1)
+	_buildOne.SetNumber(1)
+	_buildOne.SetCreated(1)
+	_buildOne.SetStatus("pending")
+
+	_buildTwo := testBuild()
+	_buildTwo.SetID(2)
+	_buildTwo.SetRepoID(1)
+	_buildTwo.SetNumber(2)
+	_buildTwo.SetCreated(2)
+	_buildTwo.SetStatus("running")
+
+	// setup types
+	_buildThree := testBuild()
+	_buildThree.SetID(3)
+	_buildThree.SetRepoID(1)
+	_buildThree.SetNumber(3)
+	_buildThree.SetCreated(1)
+	_buildThree.SetStatus("success")
+
+	_buildFour := testBuild()
+	_buildFour.SetID(4)
+	_buildFour.SetRepoID(1)
+	_buildFour.SetNumber(4)
+	_buildFour.SetCreated(5)
+	_buildFour.SetStatus("running")
+
+	_postgres, _mock := testPostgres(t)
+	defer func() { _sql, _ := _postgres.client.DB(); _sql.Close() }()
+
+	// ensure the mock expects the name query
+	_mock.ExpectExec(`UPDATE "builds" SET "status"=$1,"error"=$2,"finished"=$3,"deploy_payload"=$4 WHERE created < $5 AND (status = 'running' OR status = 'pending')`).
+		WithArgs("error", "msg", time.Now().UTC().Unix(), AnyArgument{}, 3).
```

Comment:
Is this going to create a flaky test? Because timestamp is calculated both in the test and in the code under test.

Ditto for the step and service clean tests.

Reply:
It would if we were comparing the objects, but all we do is check the count in the test